src/main/java/com/hubspot/imap/utils/parsers/string/AtomOrStringParser.java

```diff
@@ -67,6 +72,56 @@ public String parse(ByteBuf buffer) {
     return seq.toString();
   }
 
+  /**
+   * See https://tools.ietf.org/html/rfc3501.html#section-4.3
+   *
+   * String literals have the form
+   *
+   * {10}
+   * abcdefghij
+   *
+   * Where {10} represents the length of the string literal. The literal
+   * begins after any CLRF characters following the '}' character.
+   */
+  private String parseLiteral(ByteBuf buffer, AppendableCharSequence seq) {
+    int length = 0, digit = 0;
+
+    char c = (char) buffer.readUnsignedByte();
+    while (c != '}') {
+      if (Character.isDigit(c)) {
+        digit = Character.digit(c, 10);
+        length = (length * 10) + digit;
+        c = (char) buffer.readUnsignedByte();
+      } else {
+        throw new DecoderException(
+            String.format(
+                "Found non-digit character %c where a digit was expected",
+                c
```

Comment:
new line?

Reply:
just the way I format multiple params if they don't fix on one line. does look really weird here tho cuz it's only one character